setup.py

```diff
@@ -955,11 +989,67 @@ def build_torch_extension_v2(build_ext, options, torch_version):
     build_ext.build_extension(torch_mpi_lib_v2)
 
 
+def build_cmake(build_ext, ext, output_dir, options):
+    # CMake should come with pip requirement 'cmake', but let's verify
+
+    try:
+        import cmake
+        cmake_bin = os.path.join(cmake.CMAKE_BIN_DIR, 'cmake')
+        subprocess.check_call([ 'chmod',  '+x',  cmake_bin ])
```

Comment:
Don't think we should be doing that - it's very likely to fail if run as a user on linux.

Reply:
If we do not add the execution flag, the cmake will throw a permission denied error when trying to execute cmake binary.

Reply:
Are you sure?  I think when cmake python package is installed, installer does make cmake binary +x itself.

Reply:
When I do experiments on our container at DCA, the cmake python package can be imported but its binary is executable for any groups.

Reply:
Maybe we can only do the chmod trick after we install cmake ourselves (when we add cmake to the list of required packages), and only set the execution bit if it isn't already set (see: https://docs.python.org/2/library/os.html#os.access).  We can also use Python's API for setting file permissions to move towards a cross-platform solution.

Reply:
I see. After some experiments, I found that directly calling `pip install cmake` inside the setup.py can also work around the problem. Is that acceptable?

Reply:
I think a combination of os.access() & os.chmod() would be better.